util/validator.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/frain-dev/convoy/config/algo"
+)
+
+func Validate(dst interface{}) error {
+	_, err := govalidator.ValidateStruct(dst)
+
+	var messages []string
+
+	if err != nil {
+		errs := govalidator.ErrorsByField(err)
+		for field, message := range errs {
+			messages = append(messages, fmt.Sprintf("%s:%s", field, message))
+		}
+
+		validationMessages := strings.Join(messages, ", ")
+
+		return errors.New(validationMessages)
+	}
+
+	return nil
+}
+
+func init() {
+	govalidator.TagMap["supported_hash"] = govalidator.Validator(func(hash string) bool {
+		if _, ok := algo.M[hash]; !ok {
+			return false
+		}
+
+		return true
+	})
```

Comment:
I think this can be somewhere in main.go

Reply:
I thought about doing that too, but my reason for doing this is just incase we need to add more custom validation down the line, it's cleaner and clearer this way. Any downsides to doing it this way? @danvixent

Reply:
I prefer it here, let's not litter main with so much logic.

Reply:
This is probably fine.